### ISSUE TITLE
Number and Date formats

### ISSUE THREAD

**Issue by robbytobby** [westonganger/spreadsheet_architect#3]
First of all: Nice gem, makes generating spreadsheets easy with only a few lines of code. Thanks.

I've got some issues with formatting though:

**in .xlsx and .ods:**
when outputting a localized date (e.g 31.1.2010)  the cell value is preceded with a single quote ('). So I cannot work with any date functions of LibreOffice on that value.
if I do not localize the value, I get the integer representation. I admit, I can just format that column as date manually in LibreOffice, but most users will not see that little workaround.

**in .ods only:**
when outputting an integer, it is as well preceded with a " ' ", so the cell is a text value. When outputting a float, the cell uses a dot as the decimal sign, which leads to the problem, that the value is interpreted as text in (german) LibreOffice. 
Note: both integer and float work in .xlsx spreadsheets.

ps: 
spreadsheet_architect 1.2.0
axlsx 2.1.0.pre

**Comment by westonganger** [comment]
I have been attempting to add support for types (:string or :number) which is successful and I think will be in the next release.

As for the date I have been having troubles getting `axlsx` to output anything except the integer representation of the Date or Time. In ODS I had date and time working mostly but we now run into issues with date formatting which could get pretty specific to xlsx or ods formats. If someone creates a pull request to add full support for dates & formatting for both ods & xlsx that would be great. But I don't want to add a half baked solution to this.

**Comment by westonganger** [comment]
The numeric cells should now be formatted correctly in v1.2.2

**Comment by robbytobby** [comment]
First of all, thanks for your fast response.

I just tested v1.2.2 and at least on my system (Linux with LibreOffice) numeric values are now allways text, even in an xls export.  
This my be partly related to a german LibreOffice, where decimal seperator is ',' and floats are (of course) exported with a '.' 
But integer values are now preceded with a hyphen even in xls and ods

**Comment by westonganger** [comment]
I just released v1.2.3 which fixes the cell type logic. So the integers should be working correctly. I could possibly make a branch where you can test out manual cell types which you could use to test your german floats and dates if you want to play around with it.

**Comment by westonganger** [comment]
I created a branch that supports manual cell types. The supported types are:
`:date, :time, :integer, :float, :string`

Feel free to test this out and see if it fixes any of your problems and as a base point to create a pull request for this feature if you are so inclined.

**Comment by robbytobby** [comment]
Just tested the new versions

## on branch manual_cell_types:
### with format xls:

- integer: works
- float: works (no problems with german format)
- string: works
- date: does not work - resulting in integer representation

### with format ods:

- integer: does not work - empty cells
- float: works (no problems with german format)
- string: works
- date: works (shows date_time)

## on branch master
### with format xls or ods
 - integer: does not work => preceded with " ' " e.g. '4
 - float: does not work => preceded with " ' " e.g. '1.2 ; even with english localization in LibreOffice
 - date: does not work => comes as string in english format e.g. 2015-12-30

**Comment by westonganger** [comment]
For me on master using LibreoOffice on OSX with ods and xlsx I am not getting the `'` on integers or floats numbers.

**Comment by westonganger** [comment]
Are you using the defining `spreadsheet_columns` method or just using default active record columns?

**Comment by robbytobby** [comment]
I'm using defined spreadsheet_columns, but all the numeric and date values are just database attributes.
But it just came to my mind, that actually my "floats" aren't floats but decimals. But that woud not explain the integer problem. Gonna try it with default active record columns tomorrow.

**Comment by robbytobby** [comment]
Sorry for the delay. I just did set up a [new rails app](https://github.com/robbytobby/spreadsheet_test)(ruby-2.2.3, rails 4.2.6, sqlite, spreadsheet_architect-1.2.3) with just a standard scaffold for testing.  Here's what I found:

Everything works well (in .ods and .xlsx, with german and english localization) for all numeric values (integer, float, decimal), as long as I'm not defining spreadsheet_columns. As soon as I define spreadsheet_columns, all numeric values are strings in the spreadsheets, preceded with a `'`.
With both methds dates come as strings in english format.

When using the branch manual_cell_types, the problem becomes even more interesting:
If I do not define spreadsheed_columns, ods-export is rather broken: Integers result in empty cells, floats and even dates are exported correctly. xlsx-export fails with:
`Invalid Data: decimal. Cell.type must be one of [:date, :time, :float, :integer, :string, :boolean, :iso_8601].`

If I define spreadsheet colums without setting the type explicitly, all numeric values are strings again, and date is again a string in english format (Integers are not empty) for ods and xlsx.
If I set the type for each value explicitly, xlsx-export works fine for all numeric values, but date switches to the integer representation. Ods-export delivers empty cells for integers, floats and decimals are handled correctly, even the date is a date.

That's my findings so far. Just tell me if you need more information.

**Comment by westonganger** [comment]
Can you include some samples of your `spreadsheet_columns` methods that you used for testing the various branches?

**Comment by robbytobby** [comment]
```
class Foo < ActiveRecord::Base
  include SpreadsheetArchitect

  def computed_float
    int_number.to_f / 100
  end
  
  #def spreadsheet_columns
  #  [:id, :int_number, :float_number, :decimal_number, :date, :name, :computed_float]
  #end

  #def spreadsheet_columns
  #  [
  #    ['Id', :id, :integer], 
  #    ['Int Number', :int_number, :integer], 
  #    ['Float Number', :float_number, :float],
  #    ['Decimal Number', :decimal_number, :float],
  #    ['Date', :date, :date],
  #    ['Name', :name, :string],
  #    ['Computed Float', :computed_float, :float]
  #  ]
  #end
end
```

**Comment by westonganger** [comment]
I just released v1.2.4 which should fix the determination of cell type for symbol methods ex `:computed_float`. The manual cell types branch has been updated as well with the symbol method fix and should work correctly for ODS integers. Im not sure why XLSX failed because it shouldn't be setting the `:decimal` type automatically it should convert it to `:float`, see if it works with the new version. Thanks for all your help with testing by the way.
